web/controllers/organization_membership_controller.ex

```diff
@@ -19,6 +19,17 @@ defmodule CodeCorps.OrganizationMembershipController do
     render(conn, "index.json-api", data: memberships)
   end
 
+  def index(conn, %{"role" => roles}) do
+    roles = roles |> coalesce_string
+    memberships =
+      OrganizationMembership
+      |> preload([:organization, :member])
+      |> where([p], p.role in ^roles)
+      |> Repo.all
+
+    render(conn, "index.json-api", data: memberships)
+  end
```

Comment:
It looks like the problem with this approach is that the two could be combined: you may be filtering either `ids` or `roles`.

I'm not sure what the actual requests look like that are generating the filtering, and if they're both used, but this does look like a bit of an explosion of endpoints and some duplication, as well as possibly missing combinations of request types.